### PR TITLE
Add Service Binding Support

### PR DESCRIPTION
**Pull request by ImSingee**
The PR adds support for [Service Bindings](https://developers.cloudflare.com/workers/platform/bindings/about-service-bindings/)

### CODE REVIEW

File: worker-sys/src/remote_service.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+use crate::Request as EdgeRequest;
+
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen (extends = ::js_sys::Object, js_name = Fetcher)]
+    pub type RemoteService;
```
Comment (zebp) [inline worker-sys/src/remote_service.rs:8]: We should try to either match the [workers-types](https://github.com/cloudflare/workers-types) name `Fetcher` or just use `Service`. I think we should replace all occurences of `RemoteService` with `Service`.

File: worker/src/env.rs [hunk 2]
```diff
@@ -46,6 +47,11 @@ impl Env {
     pub fn durable_object(&self, binding: &str) -> Result<ObjectNamespace> {
         self.get_binding(binding)
     }
+
+    /// Access Service bindings by the binding name configured in your wrangler.toml file.
+    pub fn remote_service(&self, binding: &str) -> Result<RemoteService> {
```
Comment (zebp) [inline worker/src/env.rs:52]: Should be named `service`

**Review by zebp — CHANGES_REQUESTED** [review]
Looks good but needs some minor tweaks. I think we need to use the name `Service` instead of `Remote service` and we need to add testing in `worker-sandbox/`.

Service bindings in Miniflare seem a little rough right now, it seems that we'll need to update the GH action to add a service binding test.

**Comment by ImSingee** [comment]
@zebp For the name, I think now (in the dashboard) the previous `Worker Script` is now totally renamed to `Service`, So I think a `Service Binding` implies the `Remote Service Binding`, which is difference from the current local binding.

I don't think my name is right, but I don't think just name it`service` is good either.

**Comment by ImSingee** [comment]
For test, I think we must use [wrangler2](https://github.com/cloudflare/wrangler2), and it seems that the `wrangler.toml` should be changed too, maybe like this (remove `type` and change `build.upload` to `main`):

```
name = "testing-rust-worker"
workers_dev = true
compatibility_date = "2021-11-27" # required
main = "build/worker/shim.js"

kv_namespaces = [
    { binding = "SOME_NAMESPACE", id = "", preview_id = "" },
    { binding = "FILE_SIZES", id = "", preview_id = "" },
]

vars = { SOME_VARIABLE = "some value" }

[durable_objects]
bindings = [{ name = "COUNTER", class_name = "Counter" }]

[build]
command = "worker-build --release"

```

And after the migration to wrangler2 is done, I think I can add a test easily.

BTW, the current support for service binding in wrangler2 is merged (in https://github.com/cloudflare/wrangler2/pull/906) but isn't release now, so maybe we need to wait their release?

**Comment by cjpatton** [comment]
miniflare 2.6.0 allegedly has bineings support: https://github.com/cloudflare/miniflare/issues/301

**Comment by sergey-melnychuk** [comment]
Friendly ping @zebp @ImSingee 
Any chance making some progress here anytime soon?
If not, any objections if I pick it put and push to an acceptable form?

**Comment by zebp** [comment]
Closing in favor of #243

**Pull request closed without merging**